src/main/java/org/apache/mesos/scheduler/api/TaskResource.java

```diff
@@ -0,0 +1,189 @@
+package org.apache.mesos.scheduler.api;
+
+import com.google.inject.Inject;
+import com.googlecode.protobuf.format.JsonFormat;
+import org.apache.mesos.Protos;
+import org.apache.mesos.scheduler.TaskKiller;
+import org.apache.mesos.state.StateStore;
+import org.apache.mesos.state.api.PropertyDeserializer;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A read-only API for accessing task and frameworkId state from persistent storage.
+ */
+@Path("/v1/tasks")
+public class TaskResource {
+
+    private static final Logger logger = LoggerFactory.getLogger(TaskResource.class);
+
+    private final StateStore stateStore;
+    private final PropertyDeserializer propertyDeserializer;
+    private final TaskKiller taskKiller;
+    private final String frameworkName;
+
+    /**
+     * Creates a new TaskResource which cannot deserialize Properties. Callers will receive a
+     * "204 NO_CONTENT" HTTP response when attempting to view the content of a property.
```

Comment:
Looks like all the property handling is still over in the `StateResource`.  Was the intent to replace the `StateResource` with the `TaskResource`?